src/core/memory.cc

```diff
@@ -1801,10 +1806,12 @@ void maybe_dump_memory_diagnostics(size_t size) {
 void on_allocation_failure(size_t size) {
     alloc_stats::increment(alloc_stats::types::failed_allocs);
 
-    maybe_dump_memory_diagnostics(size);
+    bool will_abort = !abort_on_alloc_failure_suppressed
+            && abort_on_allocation_failure.load(std::memory_order_relaxed);
+
+    maybe_dump_memory_diagnostics(size, will_abort);
 
-    if (!abort_on_alloc_failure_suppressed
-            && abort_on_allocation_failure.load(std::memory_order_relaxed)) {
+    if (will_abort) {
         seastar_logger.error("Failed to allocate {} bytes", size);
```

Comment:
There is potential for this to trigger a latent bug: while we log, we allocate memory (which is in short supply) and go into some bad situation.

Reply:
True, though the logging code does go through some hoops to do relatively few allocations.

IME in a low memory situation it is almost always a relatively large allocation that fails, since as memory pressure increases the first thing that happens (even with say 10% of memory left) is that some large allocation can't be satisfied due to fragmentation ... but there are still spans and objects in the small pools available for lots of small allocations, so I guess this succeeds.

Reply:
@avikivity - you got me thinking about this case and the result was: https://github.com/scylladb/seastar/pull/1426 which tries to ensure we don't allocate on the diagnostics logging path.